src/api/ZoweExplorerZosmfApi.ts

```diff
@@ -36,26 +41,178 @@ class ZosmfApiCommon implements ZoweExplorerApi.ICommon {
         return ZosmfUssApi.getProfileTypeName();
     }
 
-    public getSession(profile?: IProfileLoaded): Session {
+    public async getSession(profile?: IProfileLoaded): Promise<Session> {
```

Comment:
Why do you need to change getSession and getStatus structure? Do you need it in this format? Dataset Viewer are using these apis and we need to inform them about this. This will probably break their extension.

If this is not an important change, Can you put it back for now? @zdmullen Can we inform Dataset Viewer of the changes that we have for this PR.

Reply:
Yes, I will do that.

Reply:
In order to use getValidSession await must be used, since it is an asynchronous function...

Reply:
I am asking about the output format of getSession() and getStatus(). You are expecting a Promise<Session> and this can break other external extension.  This `return {status: "active", session: validateSession};` is not expected by other extensions and we need to tell other teams and they need to test it.

Reply:
If the change is really needed then you don't have to change it. we just need to involve other teams and have them test this branch as well.

Reply:
Ok let me check in more detail, I was on vacation